src/datanode/src/heartbeat/handler.rs

```diff
@@ -77,16 +74,14 @@ impl HandlerGroupExecutor {
 impl HeartbeatResponseHandlerExecutor for HandlerGroupExecutor {
     fn handle(&self, mut ctx: HeartbeatResponseHandlerContext) -> Result<()> {
         for handler in &self.handlers {
-            if ctx.is_skip_all() {
-                break;
-            }
-
             if !handler.is_acceptable(&ctx) {
                 continue;
             }
 
-            if let Err(e) = handler.handle(&mut ctx) {
-                error!(e;"Error while handling: {:?}", ctx.response);
+            match handler.handle(&mut ctx) {
+                Ok(HandleControl::Done) => break,
+                Err(e) => error!(e;"Error while handling: {:?}", ctx.response),
+                _ => {} // continues
```

Comment:
Don't use wildcards in matching, please list all branches. If we add new variants to HandleControl, we won't forget to match them.